tests/v2/codecoverage/setuprancher/main.go

```diff
@@ -0,0 +1,511 @@
+
+		err = kwait.Poll(500*time.Millisecond, 2*time.Minute, func() (done bool, err error) {
+			nodeTemplate, err := client.Management.NodeTemplate.ByID(nodeTemplateResp.ID)
+			if err != nil {
+				return false, nil
+			} else if nodeTemplate != nil && nodeTemplate.ID == nodeTemplateResp.ID {
+				return true, nil
+			}
+			return false, nil
+		})
+
+		if err != nil {
+			return nil, err
+		}
+
+		err = kwait.Poll(500*time.Millisecond, 2*time.Minute, func() (done bool, err error) {
+			_, err = nodepools.NodePoolSetup(client, nodesAndRoles, clusterResp.ID, nodeTemplateResp.ID)
+			if err != nil {
+				return false, nil
+			}
+
+			return true, nil
+		})
+
+		if err != nil {
+			return nil, err
+		}
+		fmt.Println("before node pool poll")
+
+		opts := metav1.ListOptions{
+			FieldSelector:  "metadata.name=" + clusterResp.ID,
+			TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+		}
+		watchInterface, err := adminClient.GetManagementWatchInterface(management.ClusterType, opts)
+		if err != nil {
+			return nil, err
+		}
+
+		checkFunc := clusters.IsHostedProvisioningClusterReady
+
+		err = wait.WatchWait(watchInterface, checkFunc)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return clusterNames, nil
+}
+
+func writeToConfigFile(config UserClusterConfig) error {
+	result := map[string]UserClusterConfig{}
+	result[userClusterConfigsConfigurationFileKey] = config
+
+	yamlConfig, err := yaml.Marshal(result)
+
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile("userclusterconfig.yaml", yamlConfig, 0644)
+}
+
+func postRancherInstall(adminClient *rancher.Client) error {
+	clusterID, err := clusters.GetClusterIDByName(adminClient, clusterName)
+	if err != nil {
+		return err
+	}
+
+	steveClient, err := adminClient.Steve.ProxyDownstream(clusterID)
+	if err != nil {
+		return err
+	}
+
+	timeStamp := time.Now().Format(time.RFC3339)
+	settingEULA := v3.Setting{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "eula-agreed",
+		},
+		Default: timeStamp,
+		Value:   timeStamp,
+	}
+
+	urlSetting := &v3.Setting{}
+
+	_, err = steveClient.SteveType("management.cattle.io.setting").Create(settingEULA)
+	if err != nil {
+		return err
+	}
+
+	urlSettingResp, err := steveClient.SteveType("management.cattle.io.setting").ByID("server-url")
+	if err != nil {
+		return err
+	}
+
+	err = v1.ConvertToK8sType(urlSettingResp.JSONResp, urlSetting)
+	if err != nil {
+		return err
+	}
+
+	urlSetting.Value = fmt.Sprintf("https://%s", adminClient.RancherConfig.Host)
+
+	_, err = steveClient.SteveType("management.cattle.io.setting").Update(urlSettingResp, urlSetting)
+	if err != nil {
+		return err
+	}
+
+	userList, err := adminClient.Management.User.List(&types.ListOpts{
+		Filters: map[string]interface{}{
+			"username": "admin",
+		},
+	})
+	if err != nil {
+		return err
+	} else if len(userList.Data) == 0 {
+		return fmt.Errorf("admin user not found")
+	}
+
+	adminUser := &userList.Data[0]
+	setPasswordInput := management.SetPasswordInput{
+		NewPassword: adminPassword,
```

Comment:
Thanks for adding this to the post Rancher configuration.

And I just wonder if this `adminPassword` should be also included in the `CATTLE_TEST_CONFIG`?

Reply:
If you think we would do this a lot, I can add that. Since this is only to set the admin password after rancher deployment, I think for now we can just keep it how it is.

Reply:
I see where is this coming from now, we're good here.